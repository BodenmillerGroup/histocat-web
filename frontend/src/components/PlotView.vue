<template>
  <v-img :src="src" contain max-height="550px">
    <template v-slot:placeholder>
      <v-row class="fill-height ma-0" align="center" justify="center">
        <v-progress-circular indeterminate color="grey lighten-5" />
      </v-row>
    </template>
  </v-img>
</template>

<script lang="ts">
import { Component, Prop, Vue } from "vue-property-decorator";

@Component
export default class PlotView extends Vue {
  @Prop(String) readonly title!: string;
  @Prop(String) readonly src!: string;
}
</script>
