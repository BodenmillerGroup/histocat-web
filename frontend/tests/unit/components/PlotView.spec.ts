import { mount } from "@vue/test-utils";
import PlotView from "@/components/PlotView.vue";

import Vue from "vue";
import Vuetify from "vuetify";
Vue.use(Vuetify);

describe("PlotView.vue", () => {
  it("renders correctly", () => {
    const wrapper = mount(PlotView);
    expect(wrapper.element).toMatchSnapshot();
  });
});
