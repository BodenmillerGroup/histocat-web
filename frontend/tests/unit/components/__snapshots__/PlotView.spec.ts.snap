// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlotView.vue renders correctly 1`] = `
<div
  class="v-image v-responsive theme--light"
  style="max-height: 550px;"
>
  <transition-stub
    appear="true"
    name="fade-transition"
  >
    <div
      class="v-image__placeholder"
    >
      <div
        class="row fill-height ma-0 align-center justify-center"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          class="v-progress-circular v-progress-circular--indeterminate grey--text text--lighten-5"
          role="progressbar"
          style="height: 32px; width: 32px;"
        >
          <svg
            style="transform: rotate(0deg);"
            viewBox="22.857142857142858 22.857142857142858 45.714285714285715 45.714285714285715"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              class="v-progress-circular__overlay"
              cx="45.714285714285715"
              cy="45.714285714285715"
              fill="transparent"
              r="20"
              stroke-dasharray="125.664"
              stroke-dashoffset="125.66370614359172px"
              stroke-width="5.714285714285714"
            />
          </svg>
          <div
            class="v-progress-circular__info"
          />
        </div>
      </div>
    </div>
  </transition-stub>
  <div
    class="v-responsive__content"
  />
</div>
`;
